src/aggregate.d

```diff
@@ -270,13 +276,15 @@ extern (C++) abstract class AggregateDeclaration : ScopeDsymbol
             return 0;
         }
 
-        fields.setDim(0);
-
         for (size_t i = 0; i < members.dim; i++)
         {
             auto s = (*members)[i];
             if (s.apply(&func, cast(void*)this))
+            {
+                if (sizeok != SIZEOKnone)
+                    return true;
```

Comment:
> this has been broken since 2.071!

The struct layout of `S1` for this particular unittest is correct for 2.071.2 (`pragma(msg, typeof(S1.tupleof));` => `(int, void*, void*)`) - for both DMD and LDC 1.1. So I only hit an appropriate LDC assertion when merging 2.072.2 (DMD fails too).

> Typical LDC not upstreaming fixes. :-)

I came up with a quick workaround and contacted Rainer wrt. upstreamability. If you manage to fix the issue at its root, that'd be much preferred over checking all existing fields for duplicates whenever appending a new one (which is already used in https://github.com/dlang/dmd/blob/master/src/declaration.d#L1884 btw).